modules/analysis-common/build.gradle

```diff
@@ -30,6 +28,12 @@ dependencies {
   clusterModules project(':modules:mapper-extras')
 }
 
+tasks.named("yamlRestTest").configure { task->
+  if (BuildParams.getRuntimeJavaVersion().majorVersion.toInteger() < 21) {
+    task.skipTest("analysis-common/50_char_filters/pattern_replace error handling (too complex pattern)", "Requires Java 21 or later")
```

Comment:
`skipTest` doesn't exist on this task. That method is only on compat task transform tasks so this will fail. You'll want to use the blacklist system property instead. Example: https://github.com/elastic/elasticsearch/blob/510a56bb96fac0972aa578f5db0bfe5feee45c0d/docs/build.gradle#L41